core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -399,12 +402,20 @@ class LogValidatorTest {
     assertEquals(i, offsetCounter.value);
     assertEquals(now + 1, validatingResults.maxTimestampMs,
       s"Max timestamp should be ${now + 1}")
-    assertEquals(1, validatingResults.shallowOffsetOfMaxTimestampMs,
+
+    val expectedShallowOffsetOfMaxTimestamp = if (magic >= RecordVersion.V2.value) {
```

Comment:
Currently we have up to V2, if this behavior changes in the future, it could break the test? How about making it just equals V2?

Reply:
If the behavior changes, breaking the test seems correct? Very unlikely we would ever go back to a non-batching format in any case, so this seems like a reasonable default assertion.